src/main/java/org/jabref/logic/search/SearchQuery.java

```diff
@@ -42,7 +42,7 @@ public SearchQuery(String query, EnumSet<SearchRules.SearchFlags> searchFlags) {
 
         MultiFieldQueryParser queryParser = new MultiFieldQueryParser(fieldsToSearchArray, new EnglishStemAnalyzer(), boosts);
         queryParser.setAllowLeadingWildcard(true);
-        if (!query.contains("\"") && !query.contains(":") && !query.contains("*") && !query.contains("~")) {
+        if (!query.contains("\"") && !query.contains(":") && !query.contains("*") && !query.contains("~") & query.length() > 0) {
```

Comment:
just a simple optimization: Put the length check at the beginning, so  we not need to evaluate the whole expressions